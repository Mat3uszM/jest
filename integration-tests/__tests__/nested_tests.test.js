/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */
'use strict';

const runJest = require('../runJest');

test('runs tests and suites defined inside other tests', () => {
  const result = runJest('nested-tests');
  expect(result.stdout).toMatch('allSuitesCount: 16');
  expect(result.stdout).toMatch('allTestsCount: 152');
  result.stderr.split('\n').forEach(line => {
    // only tests with "failing" in the name should be failing
    if (line.includes('✕ ')) {
      expect(line).toMatch('failing');
    } else if (line.includes('✓ ')) {
      expect(line).not.toMatch('failing');
    }
  });
  const afterAlls = result.stdout
    .split('\n')
    .filter(line => line.includes(' > afterAll'));
  expect(afterAlls).toHaveLength(24);
  expect(result.stderr).toMatch('48 failed, 104 passed, 152 total');
  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toBe(1);
});
