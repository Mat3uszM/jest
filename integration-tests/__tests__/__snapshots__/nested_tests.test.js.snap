// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs tests and suites defined inside other tests 1`] = `
"  nested-tests.test.js:30
    a:sync > beforeAll > sync:passing

  nested-tests.test.js:30
    a:sync > beforeAll > sync:failing

  nested-tests.test.js:30
    a:sync > test1 > sync:passing

  nested-tests.test.js:30
    a:sync > test2 > sync:failing

  nested-tests.test.js:86
    a:sync > afterAll

  nested-tests.test.js:18
    b:async > beforeAll > async,passing

  nested-tests.test.js:18
    b:async > beforeAll > async,failing

  nested-tests.test.js:18
    b:async > test1 > async,passing

  nested-tests.test.js:18
    b:async > test2 > async,failing

  nested-tests.test.js:86
    b:async > afterAll

  nested-tests.test.js:30
    c:both:sync > beforeAll > sync:passing

  nested-tests.test.js:30
    c:both:sync > beforeAll > sync:failing

  nested-tests.test.js:18
    c:both:async > beforeAll > async,passing

  nested-tests.test.js:18
    c:both:async > beforeAll > async,failing

  nested-tests.test.js:30
    c:both:sync > test1 > sync:passing

  nested-tests.test.js:30
    c:both:sync > test2 > sync:failing

  nested-tests.test.js:18
    c:both:async > test1 > async,passing

  nested-tests.test.js:18
    c:both:async > test2 > async,failing

  nested-tests.test.js:86
    c:both:async > afterAll

  nested-tests.test.js:86
    c:both:sync > afterAll

  nested-tests.test.js:30
    d:withNested:sync > beforeAll > sync:passing

  nested-tests.test.js:30
    d:withNested:sync > beforeAll > sync:failing

  nested-tests.test.js:30
    beforeAll > describeSync > beforeAll > sync:passing

  nested-tests.test.js:30
    beforeAll > describeSync > beforeAll > sync:failing

  nested-tests.test.js:30
    beforeAll > describeSync > test1 > sync:passing

  nested-tests.test.js:30
    beforeAll > describeSync > test2 > sync:failing

  nested-tests.test.js:86
    beforeAll > describeSync > afterAll

  nested-tests.test.js:30
    beforeAll > describeAsync > beforeAll > sync:passing

  nested-tests.test.js:30
    beforeAll > describeAsync > beforeAll > sync:failing

  nested-tests.test.js:30
    beforeAll > describeAsync > test1 > sync:passing

  nested-tests.test.js:30
    beforeAll > describeAsync > test2 > sync:failing

  nested-tests.test.js:86
    beforeAll > describeAsync > afterAll

  nested-tests.test.js:30
    d:withNested:sync > test1 > sync:passing

  nested-tests.test.js:30
    d:withNested:sync > test2 > sync:failing

  nested-tests.test.js:30
    d:withNested:sync > describeSync > beforeAll > sync:passing

  nested-tests.test.js:30
    d:withNested:sync > describeSync > beforeAll > sync:failing

  nested-tests.test.js:30
    d:withNested:sync > describeSync > test1 > sync:passing

  nested-tests.test.js:30
    d:withNested:sync > describeSync > test2 > sync:failing

  nested-tests.test.js:86
    d:withNested:sync > describeSync > afterAll

  nested-tests.test.js:30
    d:withNested:sync > describeAsync > beforeAll > sync:passing

  nested-tests.test.js:30
    d:withNested:sync > describeAsync > beforeAll > sync:failing

  nested-tests.test.js:30
    d:withNested:sync > describeAsync > test1 > sync:passing

  nested-tests.test.js:30
    d:withNested:sync > describeAsync > test2 > sync:failing

  nested-tests.test.js:86
    d:withNested:sync > describeAsync > afterAll

  nested-tests.test.js:86
    d:withNested:sync > afterAll

  nested-tests.test.js:18
    e:withNested:async > beforeAll > async,passing

  nested-tests.test.js:18
    e:withNested:async > beforeAll > async,failing

  nested-tests.test.js:18
    beforeAll > describeSync > beforeAll > async,passing

  nested-tests.test.js:18
    beforeAll > describeSync > beforeAll > async,failing

  nested-tests.test.js:18
    beforeAll > describeSync > test1 > async,passing

  nested-tests.test.js:18
    beforeAll > describeSync > test2 > async,failing

  nested-tests.test.js:86
    beforeAll > describeSync > afterAll

  nested-tests.test.js:18
    beforeAll > describeAsync > beforeAll > async,passing

  nested-tests.test.js:18
    beforeAll > describeAsync > beforeAll > async,failing

  nested-tests.test.js:18
    beforeAll > describeAsync > test1 > async,passing

  nested-tests.test.js:18
    beforeAll > describeAsync > test2 > async,failing

  nested-tests.test.js:86
    beforeAll > describeAsync > afterAll

  nested-tests.test.js:18
    e:withNested:async > test1 > async,passing

  nested-tests.test.js:18
    e:withNested:async > test2 > async,failing

  nested-tests.test.js:18
    e:withNested:async > describeSync > beforeAll > async,passing

  nested-tests.test.js:18
    e:withNested:async > describeSync > beforeAll > async,failing

  nested-tests.test.js:18
    e:withNested:async > describeSync > test1 > async,passing

  nested-tests.test.js:18
    e:withNested:async > describeSync > test2 > async,failing

  nested-tests.test.js:86
    e:withNested:async > describeSync > afterAll

  nested-tests.test.js:18
    e:withNested:async > describeAsync > beforeAll > async,passing

  nested-tests.test.js:18
    e:withNested:async > describeAsync > beforeAll > async,failing

  nested-tests.test.js:18
    e:withNested:async > describeAsync > test1 > async,passing

  nested-tests.test.js:18
    e:withNested:async > describeAsync > test2 > async,failing

  nested-tests.test.js:86
    e:withNested:async > describeAsync > afterAll

  nested-tests.test.js:86
    e:withNested:async > afterAll

  nested-tests.test.js:30
    f:withNested:both:sync > beforeAll > sync:passing

  nested-tests.test.js:30
    f:withNested:both:sync > beforeAll > sync:failing

  nested-tests.test.js:30
    beforeAll > describeSync > beforeAll > sync:passing

  nested-tests.test.js:30
    beforeAll > describeSync > beforeAll > sync:failing

  nested-tests.test.js:30
    beforeAll > describeSync > test1 > sync:passing

  nested-tests.test.js:30
    beforeAll > describeSync > test2 > sync:failing

  nested-tests.test.js:86
    beforeAll > describeSync > afterAll

  nested-tests.test.js:30
    beforeAll > describeAsync > beforeAll > sync:passing

  nested-tests.test.js:30
    beforeAll > describeAsync > beforeAll > sync:failing

  nested-tests.test.js:30
    beforeAll > describeAsync > test1 > sync:passing

  nested-tests.test.js:30
    beforeAll > describeAsync > test2 > sync:failing

  nested-tests.test.js:86
    beforeAll > describeAsync > afterAll

  nested-tests.test.js:18
    f:withNested:both:async > beforeAll > async,passing

  nested-tests.test.js:18
    f:withNested:both:async > beforeAll > async,failing

  nested-tests.test.js:18
    beforeAll > describeSync > beforeAll > async,passing

  nested-tests.test.js:18
    beforeAll > describeSync > beforeAll > async,failing

  nested-tests.test.js:18
    beforeAll > describeSync > test1 > async,passing

  nested-tests.test.js:18
    beforeAll > describeSync > test2 > async,failing

  nested-tests.test.js:86
    beforeAll > describeSync > afterAll

  nested-tests.test.js:18
    beforeAll > describeAsync > beforeAll > async,passing

  nested-tests.test.js:18
    beforeAll > describeAsync > beforeAll > async,failing

  nested-tests.test.js:18
    beforeAll > describeAsync > test1 > async,passing

  nested-tests.test.js:18
    beforeAll > describeAsync > test2 > async,failing

  nested-tests.test.js:86
    beforeAll > describeAsync > afterAll

  nested-tests.test.js:30
    f:withNested:both:sync > test1 > sync:passing

  nested-tests.test.js:30
    f:withNested:both:sync > test2 > sync:failing

  nested-tests.test.js:30
    f:withNested:both:sync > describeSync > beforeAll > sync:passing

  nested-tests.test.js:30
    f:withNested:both:sync > describeSync > beforeAll > sync:failing

  nested-tests.test.js:30
    f:withNested:both:sync > describeSync > test1 > sync:passing

  nested-tests.test.js:30
    f:withNested:both:sync > describeSync > test2 > sync:failing

  nested-tests.test.js:86
    f:withNested:both:sync > describeSync > afterAll

  nested-tests.test.js:30
    f:withNested:both:sync > describeAsync > beforeAll > sync:passing

  nested-tests.test.js:30
    f:withNested:both:sync > describeAsync > beforeAll > sync:failing

  nested-tests.test.js:30
    f:withNested:both:sync > describeAsync > test1 > sync:passing

  nested-tests.test.js:30
    f:withNested:both:sync > describeAsync > test2 > sync:failing

  nested-tests.test.js:86
    f:withNested:both:sync > describeAsync > afterAll

  nested-tests.test.js:18
    f:withNested:both:async > test1 > async,passing

  nested-tests.test.js:18
    f:withNested:both:async > test2 > async,failing

  nested-tests.test.js:18
    f:withNested:both:async > describeSync > beforeAll > async,passing

  nested-tests.test.js:18
    f:withNested:both:async > describeSync > beforeAll > async,failing

  nested-tests.test.js:18
    f:withNested:both:async > describeSync > test1 > async,passing

  nested-tests.test.js:18
    f:withNested:both:async > describeSync > test2 > async,failing

  nested-tests.test.js:86
    f:withNested:both:async > describeSync > afterAll

  nested-tests.test.js:18
    f:withNested:both:async > describeAsync > beforeAll > async,passing

  nested-tests.test.js:18
    f:withNested:both:async > describeAsync > beforeAll > async,failing

  nested-tests.test.js:18
    f:withNested:both:async > describeAsync > test1 > async,passing

  nested-tests.test.js:18
    f:withNested:both:async > describeAsync > test2 > async,failing

  nested-tests.test.js:86
    f:withNested:both:async > describeAsync > afterAll

  nested-tests.test.js:86
    f:withNested:both:async > afterAll

  nested-tests.test.js:86
    f:withNested:both:sync > afterAll

  nested-tests.test.js:115
    { allSuitesCount: 16, allTestsCount: 152 }

"
`;
